ci: investigate flaky test - panic during ChannelCall("setqflist")

During the test of commit https://github.com/myitcv/govim/commit/a17af79bd412876e17d6d998b6c96bb3b5e2a00f in Travis one of the jobs failed, rerunning that job made it pass.

Job:
<pre>GO_VERSION="go1.13beta1" VIM_FLAVOR="gvim" VIM_VERSION="v8.1.1657" VIM_COMMAND="xvfb-run -a gvim -f"</pre>

Failed test:
<pre>
> go install golang.org/x/tools/gopls
> rm -f $(git ls-files -- ':!:cmd/govim/internal/golang_org_x_tools' '**/gen_*.*' 'gen_*.*') .travis.yml
> go generate $(go list ./... | grep -v 'govim/internal/golang_org_x_tools')
> go test $(go list ./... | grep -v 'govim/internal/golang_org_x_tools')
ok  	github.com/myitcv/govim	10.373s
panic: driver error: ChannelCall("setqflist") failed: tomb: dying [recovered]
	panic: driver error: ChannelCall("setqflist") failed: tomb: dying [recovered]
	panic: driver error: ChannelCall("setqflist") failed: tomb: dying
goroutine 265 [running]:
github.com/myitcv/govim.(*govimImpl).goHandleShutdown.func1.1()
	/home/travis/govim/govim.go:241 +0x10a
panic(0x65ae60, 0xc00007f650)
	/go/src/runtime/panic.go:663 +0x1b2
github.com/myitcv/govim.(*govimImpl).Schedule.func1.1(0xc0000f7680, 0xc0004dc000)
	/home/travis/govim/govim.go:223 +0x184
panic(0x65ae60, 0xc00007f650)
	/go/src/runtime/panic.go:663 +0x1b2
github.com/myitcv/govim/internal/plugin.Driver.Errorf(...)
	/home/travis/govim/internal/plugin/plugin.go:61
github.com/myitcv/govim/internal/plugin.Driver.ChannelCall(0x6f7360, 0xc0004dc000, 0x6a3d04, 0x5, 0x6a4faf, 0x9, 0xc00008db20, 0x2, 0x2, 0xc00025a6c0, ...)
	/home/travis/govim/internal/plugin/plugin.go:75 +0x1c7
github.com/myitcv/govim/cmd/govim.(*vimstate).updateQuickfix(0xc0000b6820, 0x0, 0x0, 0x0, 0x0, 0xc00005c180)
	/home/travis/govim/cmd/govim/quickfix.go:100 +0xefb
github.com/myitcv/govim/cmd/govim.(*govimplugin).PublishDiagnostics.func1(0x6f7360, 0xc0004dc000, 0xc000482eb8, 0xc000598ec0)
	/home/travis/govim/cmd/govim/gopls_client.go:110 +0x62
github.com/myitcv/govim.(*govimImpl).Schedule.func1(0x0, 0x0)
	/home/travis/govim/govim.go:232 +0x91
github.com/myitcv/govim.(*govimImpl).runEventQueue.func1(0xc000598f10, 0x0)
	/home/travis/govim/govim.go:552 +0x26
github.com/myitcv/govim.(*govimImpl).goHandleShutdown.func1(0x0, 0x0)
	/home/travis/govim/govim.go:244 +0x62
gopkg.in/tomb%2ev2.(*Tomb).run(0xc0004dc090, 0xc00007f4d0)
	/home/travis/gopath/pkg/mod/gopkg.in/tomb.v2@v2.0.0-20161208151619-d5d1b5820637/tomb.go:163 +0x2b
created by gopkg.in/tomb%2ev2.(*Tomb).Go
	/home/travis/gopath/pkg/mod/gopkg.in/tomb.v2@v2.0.0-20161208151619-d5d1b5820637/tomb.go:159 +0xc7
FAIL	github.com/myitcv/govim/cmd/govim	5.786s
?   	github.com/myitcv/govim/cmd/govim/config	[no test files]
?   	github.com/myitcv/govim/cmd/govim/internal/types	[no test files]
?   	github.com/myitcv/govim/internal/plugin	[no test files]
?   	github.com/myitcv/govim/internal/queue	[no test files]
?   	github.com/myitcv/govim/testdriver	[no test files]
?   	github.com/myitcv/govim/testsetup	[no test files]
FAIL
Error on linue 34 in ./_scripts/dockerRun.sh
The command "./_scripts/runDockerRun.sh" exited with 1.
</pre>

Entire log:
<details><pre>
travis_fold:start:worker_info
[0K[33;1mWorker information[0m
hostname: 29323bfa-799b-4ebd-ae6e-611a6998ab66@1.worker-org-859d75c9bb-q629b.gce-production-2
version: v6.2.1 https://github.com/travis-ci/worker/tree/4e3246c044eb4915c2378ffacd0b3d3ed0136bba
instance: travis-job-03385c87-ca8a-4cab-b523-dc7008d82ee0 travis-ci-sardonyx-xenial-1553530528-f909ac5 (via amqp)
startup: 6.506133406s
travis_fold:end:worker_info
[0Ktravis_fold:start:system_info
[0K[33;1mBuild system information[0m
Build language: generic
Build group: stable
Build dist: xenial
Build id: 564365772
Job id: 564365784
Runtime kernel version: 4.15.0-1028-gcp
travis-build version: 24c88d12b
[34m[1mBuild image provisioning date and time[0m
Mon Mar 25 16:43:24 UTC 2019
[34m[1mOperating System Details[0m
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.6 LTS
Release:	16.04
Codename:	xenial
[34m[1mSystemd Version[0m
systemd 229
[34m[1mCookbooks Version[0m
42e42e4 https://github.com/travis-ci/travis-cookbooks/tree/42e42e4
[34m[1mgit version[0m
git version 2.21.0
[34m[1mbash version[0m
GNU bash, version 4.3.48(1)-release (x86_64-pc-linux-gnu)
[34m[1mgcc version[0m
gcc (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609
[34m[1mdocker version[0m
Client:
 Version:           18.06.0-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        0ffa825
 Built:             Wed Jul 18 19:11:02 2018
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.0-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       0ffa825
  Built:            Wed Jul 18 19:09:05 2018
  OS/Arch:          linux/amd64
  Experimental:     false
[34m[1mclang version[0m
clang version 7.0.0 (tags/RELEASE_700/final)
[34m[1mjq version[0m
jq-1.5
[34m[1mbats version[0m
Bats 0.4.0
[34m[1mshellcheck version[0m
0.6.0
[34m[1mshfmt version[0m
v2.6.3
[34m[1mccache version[0m
3.2.4
[34m[1mcmake version[0m
cmake version 3.12.4
[34m[1mheroku version[0m
heroku/7.22.7 linux-x64 node-v11.10.1
[34m[1mimagemagick version[0m
Version: ImageMagick 6.8.9-9 Q16 x86_64 2018-09-28 http://www.imagemagick.org
[34m[1mmd5deep version[0m
4.4
[34m[1mmercurial version[0m
version 4.8
[34m[1mmysql version[0m
mysql  Ver 14.14 Distrib 5.7.25, for Linux (x86_64) using  EditLine wrapper
[34m[1mopenssl version[0m
OpenSSL 1.0.2g  1 Mar 2016
[34m[1mpacker version[0m
1.3.3
[34m[1mpostgresql client version[0m
psql (PostgreSQL) 10.7 (Ubuntu 10.7-1.pgdg16.04+1)
[34m[1mragel version[0m
Ragel State Machine Compiler version 6.8 Feb 2013
[34m[1msudo version[0m
1.8.16
[34m[1mgzip version[0m
gzip 1.6
[34m[1mzip version[0m
Zip 3.0
[34m[1mvim version[0m
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Nov 24 2016 16:44:48)
[34m[1miptables version[0m
iptables v1.6.0
[34m[1mcurl version[0m
curl 7.47.0 (x86_64-pc-linux-gnu) libcurl/7.47.0 GnuTLS/3.4.10 zlib/1.2.8 libidn/1.32 librtmp/2.3
[34m[1mwget version[0m
GNU Wget 1.17.1 built on linux-gnu.
[34m[1mrsync version[0m
rsync  version 3.1.1  protocol version 31
[34m[1mgimme version[0m
v1.5.3
[34m[1mnvm version[0m
0.34.0
[34m[1mperlbrew version[0m
/home/travis/perl5/perlbrew/bin/perlbrew  - App::perlbrew/0.86
[34m[1mphpenv version[0m
rbenv 1.1.2
[34m[1mrvm version[0m
rvm 1.29.7 (latest) by Michal Papis, Piotr Kuczynski, Wayne E. Seguin [https://rvm.io]
[34m[1mdefault ruby version[0m
ruby 2.5.3p105 (2018-10-18 revision 65156) [x86_64-linux]
[34m[1mCouchDB version[0m
couchdb 1.6.1
[34m[1mElasticSearch version[0m
5.5.0
[34m[1mInstalled Firefox version[0m
firefox 63.0.1
[34m[1mMongoDB version[0m
MongoDB 4.0.7
[34m[1mPhantomJS version[0m
2.1.1
[34m[1mPre-installed PostgreSQL versions[0m
9.4.21
9.5.16
9.6.12
[34m[1mRedis version[0m
redis-server 5.0.4
[34m[1mPre-installed Go versions[0m
1.11.1
[34m[1mmvn version[0m
Apache Maven 3.6.0 (97c98ec64a1fdfee7767ce5ffb20918da4f719f3; 2018-10-24T18:41:47Z)
[34m[1mgradle version[0m
Gradle 4.10.2!
[34m[1mlein version[0m
Leiningen 2.9.1 on Java 11.0.2 OpenJDK 64-Bit Server VM
[34m[1mPre-installed Node.js versions[0m
v10.15.3
v11.0.0
v4.9.1
v6.17.0
v8.12.0
v8.15.1
v8.9
[34m[1mphpenv versions[0m
  system
  5.6
  5.6.40
  7.1
  7.1.27
  7.2
* 7.2.15 (set by /home/travis/.phpenv/version)
  hhvm
  hhvm-stable
[34m[1mcomposer --version[0m
Composer version 1.8.4 2019-02-11 10:52:10
[34m[1mPre-installed Ruby versions[0m
ruby-2.3.8
ruby-2.4.5
ruby-2.5.3
travis_fold:end:system_info
[0K
travis_fold:start:docker_mtu
[0Ktravis_fold:end:docker_mtu
[0Ktravis_fold:start:resolvconf
[0Ktravis_fold:end:resolvconf
[0K
travis_fold:start:git.checkout
[0Ktravis_time:start:00a5a0fa
[0K$ git clone --depth=50 https://github.com/myitcv/govim.git myitcv/govim
Cloning into 'myitcv/govim'...
travis_time:end:00a5a0fa:start=1564230657100996439,finish=1564230658082235426,duration=981238987
[0K$ cd myitcv/govim
travis_time:start:0da1337e
[0K$ git fetch origin +refs/pull/405/merge:
From https://github.com/myitcv/govim
 * branch            refs/pull/405/merge -> FETCH_HEAD
travis_time:end:0da1337e:start=1564230658087323673,finish=1564230658719932788,duration=632609115
[0K$ git checkout -qf FETCH_HEAD
travis_fold:end:git.checkout
[0K

[33;1mSetting environment variables from repository settings[0m
$ export GH_TOKEN=[secure]
$ export GH_USER=myitcv

[33;1mSetting environment variables from .travis.yml[0m
$ export GO111MODULE=on
$ export GOPROXY=https://proxy.golang.org
$ export GO_VERSION="go1.13beta1"
$ export VIM_FLAVOR="gvim"
$ export VIM_VERSION="v8.1.1657"
$ export VIM_COMMAND="xvfb-run -a gvim -f"

$ bash -c 'echo $BASH_VERSION'
4.3.48(1)-release

travis_fold:start:before_install.1
[0Ktravis_time:start:06123f84
[0K$ docker --version
Docker version 18.06.0-ce, build 0ffa825
travis_time:end:06123f84:start=1564230658763104251,finish=1564230659046396215,duration=283291964
[0Ktravis_fold:end:before_install.1
[0Ktravis_fold:start:before_install.2
[0Ktravis_time:start:1c92890d
[0K$ ./_scripts/buildGovimImage.sh
Sending build context to Docker daemon  2.048kB


Step 1/11 : FROM govim/govim:go1.13beta1_gvim_v8.1.1657_v1
go1.13beta1_gvim_v8.1.1657_v1: Pulling from govim/govim
898c46f3b1a1: Pulling fs layer
63366dfa0a50: Pulling fs layer
041d4cd74a92: Pulling fs layer
6e1bee0f8701: Pulling fs layer
930c33e06246: Pulling fs layer
c2eb65575a45: Pulling fs layer
0adb2cbf93a0: Pulling fs layer
0650c35c0db6: Pulling fs layer
7c859565a35f: Pulling fs layer
ce8f93ab5d45: Pulling fs layer
366a987b472f: Pulling fs layer
fd13a62bfca1: Pulling fs layer
6f6fdd200a0a: Pulling fs layer
339f8e3b8dd0: Pulling fs layer
4d7ad2f030fc: Pulling fs layer
b93072a031bb: Pulling fs layer
6e1bee0f8701: Waiting
930c33e06246: Waiting
c2eb65575a45: Waiting
0adb2cbf93a0: Waiting
0650c35c0db6: Waiting
7c859565a35f: Waiting
ce8f93ab5d45: Waiting
366a987b472f: Waiting
fd13a62bfca1: Waiting
6f6fdd200a0a: Waiting
339f8e3b8dd0: Waiting
4d7ad2f030fc: Waiting
b93072a031bb: Waiting
63366dfa0a50: Verifying Checksum
63366dfa0a50: Download complete
041d4cd74a92: Download complete
898c46f3b1a1: Verifying Checksum
898c46f3b1a1: Download complete
6e1bee0f8701: Verifying Checksum
6e1bee0f8701: Download complete
930c33e06246: Verifying Checksum
930c33e06246: Download complete
c2eb65575a45: Verifying Checksum
c2eb65575a45: Download complete
0adb2cbf93a0: Verifying Checksum
0adb2cbf93a0: Download complete
ce8f93ab5d45: Verifying Checksum
ce8f93ab5d45: Download complete
0650c35c0db6: Verifying Checksum
0650c35c0db6: Download complete
fd13a62bfca1: Verifying Checksum
fd13a62bfca1: Download complete
7c859565a35f: Verifying Checksum
7c859565a35f: Download complete
339f8e3b8dd0: Verifying Checksum
339f8e3b8dd0: Download complete
6f6fdd200a0a: Verifying Checksum
6f6fdd200a0a: Download complete
898c46f3b1a1: Pull complete
63366dfa0a50: Pull complete
041d4cd74a92: Pull complete
b93072a031bb: Verifying Checksum
b93072a031bb: Download complete
6e1bee0f8701: Pull complete
930c33e06246: Pull complete
366a987b472f: Verifying Checksum
366a987b472f: Download complete
c2eb65575a45: Pull complete
4d7ad2f030fc: Verifying Checksum
4d7ad2f030fc: Download complete
0adb2cbf93a0: Pull complete
0650c35c0db6: Pull complete
7c859565a35f: Pull complete
ce8f93ab5d45: Pull complete
366a987b472f: Pull complete
fd13a62bfca1: Pull complete
6f6fdd200a0a: Pull complete
339f8e3b8dd0: Pull complete
4d7ad2f030fc: Pull complete
b93072a031bb: Pull complete
Digest: sha256:72676fe0cf02c444ead081fd624a5409a79fc33a0459afb9524e13650cf7bb74
Status: Downloaded newer image for govim/govim:go1.13beta1_gvim_v8.1.1657_v1
 ---> 7c78ad05d901
Step 2/11 : ARG USER
 ---> Running in 2614b3aa7a3b
Removing intermediate container 2614b3aa7a3b
 ---> 93769f49cb46
Step 3/11 : ARG UID
 ---> Running in 2698617fd11e
Removing intermediate container 2698617fd11e
 ---> bd13c6f5369a
Step 4/11 : ARG GID
 ---> Running in d50762043726
Removing intermediate container d50762043726
 ---> 6256ad2f46a3
Step 5/11 : ENV PATH=/vbash/bin:/home/$USER/.local/bin:$PATH
 ---> Running in 584ae881df0f
Removing intermediate container 584ae881df0f
 ---> 31de0bf95bed
Step 6/11 : ENV GOPATH=/home/$USER/gopath
 ---> Running in 06c8a56799bd
Removing intermediate container 06c8a56799bd
 ---> 9287f6a052f4
Step 7/11 : RUN sh -c "if ! getent group $GID; then groupadd -g $GID $USER; fi" &&     adduser --uid $UID --gid $GID --disabled-password --gecos "" $USER
 ---> Running in 7453eda639da
Adding user `travis' ...
Adding new user `travis' (2000) with group `travis' ...
Creating home directory `/home/travis' ...
Copying files from `/etc/skel' ...
Removing intermediate container 7453eda639da
 ---> 0d6c43b43d44
Step 8/11 : RUN usermod -aG sudo $USER
 ---> Running in a397306c3465
Removing intermediate container a397306c3465
 ---> 739976e1d5b4
Step 9/11 : RUN echo "$USER ALL=(ALL:ALL) NOPASSWD: ALL" > /etc/sudoers.d/$USER
 ---> Running in 5b460542b7c0
Removing intermediate container 5b460542b7c0
 ---> f1838ea85ae1
Step 10/11 : WORKDIR /home/$USER
 ---> Running in cf4eed50ae27
Removing intermediate container cf4eed50ae27
 ---> d712114be9fa
Step 11/11 : USER $USER
 ---> Running in 6662f36bf34d
Removing intermediate container 6662f36bf34d
 ---> 1b8f68aef265
Successfully built 1b8f68aef265
Successfully tagged govim:latest
travis_time:end:1c92890d:start=1564230659051608323,finish=1564230746078539787,duration=87026931464
[0Ktravis_fold:end:before_install.2
[0Ktravis_time:start:14387d4e
[0K$ ./_scripts/runDockerRun.sh
> source "${BASH_SOURCE%/*}/common.bash"
> eval "VIM_COMMAND=\"\$DEFAULT_${VIM_FLAVOR^^}_COMMAND\""
> cat <<EOD
Environment is:
  VIM_FLAVOR=$VIM_FLAVOR
  VIM_COMMAND=$VIM_COMMAND
EOD
Environment is:
  VIM_FLAVOR=gvim
  VIM_COMMAND=xvfb-run -a gvim -f
> echo -e "machine api.github.com\n  login $GH_USER\n  password $GH_TOKEN" >>~/.netrc
> echo -e "machine github.com\n  login $GH_USER\n  password $GH_TOKEN" >>~/.netrc
> echo -e "machine githubusercontent.com\n  login $GH_USER\n  password $GH_TOKEN" >>~/.netrc
> go version
go version go1.13beta1 linux/amd64
> vim --version
VIM - Vi IMproved 8.1 (2018 May 18, compiled Jul 18 2019 08:15:12)
Included patches: 1-1657
Compiled by root@63b25e28bbf7
Huge version with GTK2 GUI.  Features included (+) or not (-):
+acl               -farsi             -mouse_sysmouse    -tag_any_white
+arabic            +file_in_path      +mouse_urxvt       -tcl
+autocmd           +find_in_path      +mouse_xterm       +termguicolors
+autochdir         +float             +multi_byte        +terminal
-autoservername    +folding           +multi_lang        +terminfo
+balloon_eval      -footer            -mzscheme          +termresponse
+balloon_eval_term +fork()            -netbeans_intg     +textobjects
+browse            +gettext           +num64             +textprop
++builtin_terms    -hangul_input      +packages          +timers
+byte_offset       +iconv             +path_extra        +title
+channel           +insert_expand     -perl              +toolbar
+cindent           +job               +persistent_undo   +user_commands
+clientserver      +jumplist          +postscript        +vartabs
+clipboard         +keymap            +printer           +vertsplit
+cmdline_compl     +lambda            +profile           +virtualedit
+cmdline_hist      +langmap           -python            +visual
+cmdline_info      +libcall           -python3           +visualextra
+comments          +linebreak         +quickfix          +viminfo
+conceal           +lispindent        +reltime           +vreplace
+cryptv            +listcmds          +rightleft         +wildignore
+cscope            +localmap          -ruby              +wildmenu
+cursorbind        -lua               +scrollbind        +windows
+cursorshape       +menu              +signs             +writebackup
+dialog_con_gui    +mksession         +smartindent       +X11
+diff              +modify_fname      -sound             -xfontset
+digraphs          +mouse             +spell             +xim
+dnd               +mouseshape        +startuptime       -xpm
-ebcdic            +mouse_dec         +statusline        +xsmp_interact
+emacs_tags        -mouse_gpm         -sun_workshop      +xterm_clipboard
+eval              -mouse_jsbterm     +syntax            -xterm_save
+ex_extra          +mouse_netterm     +tag_binary        
+extra_search      +mouse_sgr         -tag_old_static    
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "$VIM/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
2nd user gvimrc file: "~/.vim/gvimrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/vim/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK  -pthread -I/usr/include/gtk-2.0 -I/usr/lib/x86_64-linux-gnu/gtk-2.0/include -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/libpng16   -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1       
Linking: gcc   -L/usr/local/lib -Wl,--as-needed -o vim   -lgtk-x11-2.0 -lgdk-x11-2.0 -lpangocairo-1.0 -latk-1.0 -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lpangoft2-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lfontconfig -lfreetype -lSM -lICE -lXt -lX11 -lXdmcp -lSM -lICE  -lm -ltinfo -lnsl  -ldl           
> ./_scripts/revendorToolsInternal.sh
	> source "${BASH_SOURCE%/*}/common.bash"
	> shopt -s globstar
	> shopt -s extglob
	> go mod download
	go: finding github.com/BurntSushi/toml v0.3.1
	go: finding github.com/acarl005/stripansi v0.0.0-20180116102854-5a71ef0e047d
	go: finding github.com/frankban/quicktest v1.2.2
	go: finding github.com/fsnotify/fsnotify v1.4.7
	go: finding github.com/google/go-cmp v0.2.1-0.20190312032427-6f77996f0c42
	go: finding github.com/google/renameio v0.1.0
	go: finding github.com/kisielk/gotool v1.0.0
	go: finding github.com/kr/pretty v0.1.0
	go: finding github.com/kr/pty v1.1.5
	go: finding github.com/kr/text v0.1.0
	go: finding github.com/myitcv/vbash v0.0.4
	go: finding github.com/rogpeppe/clock v0.0.0-20190514195947-2896927a307a
	go: finding github.com/rogpeppe/go-internal v1.3.0
	go: finding golang.org/x/crypto v0.0.0-20190510104115-cbcb75029529
	go: finding golang.org/x/mod v0.0.0-20190513183733-4bf6d317e70e
	go: finding golang.org/x/net v0.0.0-20190620200207-3b0461eec859
	go: finding golang.org/x/sync v0.0.0-20190423024810-112230192c58
	go: finding golang.org/x/sys v0.0.0-20190429190828-d89cdac9e872
	go: finding golang.org/x/text v0.3.0
	go: finding golang.org/x/tools v0.0.0-20190716221150-e98af2309876
	go: finding golang.org/x/tools/gopls v0.1.4-0.20190716221150-e98af2309876
	go: finding gopkg.in/check.v1 v1.0.0-20180628173108-788fd7840127
	go: finding gopkg.in/errgo.v2 v2.1.0
	go: finding gopkg.in/retry.v1 v1.0.3
	go: finding gopkg.in/tomb.v2 v2.0.0-20161208151619-d5d1b5820637
	go: finding honnef.co/go/tools v0.0.0-20190614002413-cb51c254f01b
	go: finding mvdan.cc/sh/v3 v3.0.0-alpha1
	> tools=$(go list -m -f={{.Dir}} golang.org/x/tools)
	> echo "Tools is $tools"
	Tools is /home/travis/gopath/pkg/mod/golang.org/x/tools@v0.0.0-20190716221150-e98af2309876
	> cd $(git rev-parse --show-toplevel)
	> rsync -a --delete --chmod=D0755,F0644 $tools/internal/ ./cmd/govim/internal/golang_org_x_tools
	> sed -i 's+golang.org/x/tools/internal+github.com/myitcv/govim/cmd/govim/internal/golang_org_x_tools+g' ./cmd/govim/internal/golang_org_x_tools/**/*.go
	> rm ./cmd/govim/internal/golang_org_x_tools/**/*_test.go
	> rm -f ./cmd/govim/internal/golang_org_x_tools/LICENSE
	> cp $tools/LICENSE ./cmd/govim/internal/golang_org_x_tools
	> go mod tidy
> go install golang.org/x/tools/gopls
> rm -f $(git ls-files -- ':!:cmd/govim/internal/golang_org_x_tools' '**/gen_*.*' 'gen_*.*') .travis.yml
> go generate $(go list ./... | grep -v 'govim/internal/golang_org_x_tools')
> go test $(go list ./... | grep -v 'govim/internal/golang_org_x_tools')
ok  	github.com/myitcv/govim	10.373s
panic: driver error: ChannelCall("setqflist") failed: tomb: dying [recovered]
	panic: driver error: ChannelCall("setqflist") failed: tomb: dying [recovered]
	panic: driver error: ChannelCall("setqflist") failed: tomb: dying

goroutine 265 [running]:
github.com/myitcv/govim.(*govimImpl).goHandleShutdown.func1.1()
	/home/travis/govim/govim.go:241 +0x10a
panic(0x65ae60, 0xc00007f650)
	/go/src/runtime/panic.go:663 +0x1b2
github.com/myitcv/govim.(*govimImpl).Schedule.func1.1(0xc0000f7680, 0xc0004dc000)
	/home/travis/govim/govim.go:223 +0x184
panic(0x65ae60, 0xc00007f650)
	/go/src/runtime/panic.go:663 +0x1b2
github.com/myitcv/govim/internal/plugin.Driver.Errorf(...)
	/home/travis/govim/internal/plugin/plugin.go:61
github.com/myitcv/govim/internal/plugin.Driver.ChannelCall(0x6f7360, 0xc0004dc000, 0x6a3d04, 0x5, 0x6a4faf, 0x9, 0xc00008db20, 0x2, 0x2, 0xc00025a6c0, ...)
	/home/travis/govim/internal/plugin/plugin.go:75 +0x1c7
github.com/myitcv/govim/cmd/govim.(*vimstate).updateQuickfix(0xc0000b6820, 0x0, 0x0, 0x0, 0x0, 0xc00005c180)
	/home/travis/govim/cmd/govim/quickfix.go:100 +0xefb
github.com/myitcv/govim/cmd/govim.(*govimplugin).PublishDiagnostics.func1(0x6f7360, 0xc0004dc000, 0xc000482eb8, 0xc000598ec0)
	/home/travis/govim/cmd/govim/gopls_client.go:110 +0x62
github.com/myitcv/govim.(*govimImpl).Schedule.func1(0x0, 0x0)
	/home/travis/govim/govim.go:232 +0x91
github.com/myitcv/govim.(*govimImpl).runEventQueue.func1(0xc000598f10, 0x0)
	/home/travis/govim/govim.go:552 +0x26
github.com/myitcv/govim.(*govimImpl).goHandleShutdown.func1(0x0, 0x0)
	/home/travis/govim/govim.go:244 +0x62
gopkg.in/tomb%2ev2.(*Tomb).run(0xc0004dc090, 0xc00007f4d0)
	/home/travis/gopath/pkg/mod/gopkg.in/tomb.v2@v2.0.0-20161208151619-d5d1b5820637/tomb.go:163 +0x2b
created by gopkg.in/tomb%2ev2.(*Tomb).Go
	/home/travis/gopath/pkg/mod/gopkg.in/tomb.v2@v2.0.0-20161208151619-d5d1b5820637/tomb.go:159 +0xc7
FAIL	github.com/myitcv/govim/cmd/govim	5.786s
?   	github.com/myitcv/govim/cmd/govim/config	[no test files]
?   	github.com/myitcv/govim/cmd/govim/internal/types	[no test files]
?   	github.com/myitcv/govim/internal/plugin	[no test files]
?   	github.com/myitcv/govim/internal/queue	[no test files]
?   	github.com/myitcv/govim/testdriver	[no test files]
?   	github.com/myitcv/govim/testsetup	[no test files]
FAIL
Error on linue 34 in ./_scripts/dockerRun.sh
travis_time:end:14387d4e:start=1564230746085863169,finish=1564230777985497724,duration=31899634555
[0K[31;1mThe command "./_scripts/runDockerRun.sh" exited with 1.[0m



Done. Your build exited with 1.
</pre></details>

Also happens in `script-bad_complete`:
<pre>
Waiting for govim driver connection on 0.0.0.0:44759...
2019-07-27T20:07:07.152266_NGA9QZ: vim start =======================
2019-07-27T20:07:07.152266_NGA9QZ: sendJSONMsg: [0,[42,"ex","qall!"]]
2019-07-27T20:07:07.152266_NGA9QZ: vim end =======================
2019-07-27T20:07:07.152407_NGA9QZ: vim start =======================
2019-07-27T20:07:07.152407_NGA9QZ: recvJSONMsg: [45] ["callback",41,["","/tmp/go-test-script744136382/script-bad_complete"]]
2019-07-27T20:07:07.152407_NGA9QZ: vim end =======================
2019-07-27T20:07:07.152439_NGA9QZ: run: waiting to read a JSON message
2019-07-27T20:07:07.152794_NGA9QZ: vim start =======================
2019-07-27T20:07:07.152794_NGA9QZ: closing connection
2019-07-27T20:07:07.152794_NGA9QZ: vim end =======================
2019-07-27T20:07:07.152818_NGA9QZ: gopls server start =======================
2019-07-27T20:07:07.152818_NGA9QZ: gopls.Shutdown() call
2019-07-27T20:07:07.152818_NGA9QZ: gopls server end =======================
2019-07-27T20:07:07.152911_NGA9QZ: vim start =======================
2019-07-27T20:07:07.152911_NGA9QZ: sendJSONMsg: [0,[43,"call","setqflist",[{"filename":"main.go","lnum":6,"col":1,"text":"expected selector or type assertion, found '}'","buf":1}],"r"]]
2019-07-27T20:07:07.152911_NGA9QZ: vim end =======================
panic: driver error: ChannelCall("setqflist") failed: tomb: dying [recovered]
	panic: driver error: ChannelCall("setqflist") failed: tomb: dying [recovered]
	panic: driver error: ChannelCall("setqflist") failed: tomb: dying

goroutine 554 [running]:
github.com/myitcv/govim.(*govimImpl).goHandleShutdown.func1.1()
	/home/pontus/govim/govim.go:256 +0x10a
panic(0x650080, 0xc0007b01e0)
	/go/src/runtime/panic.go:522 +0x1b5
github.com/myitcv/govim.(*govimImpl).Schedule.func1.1(0xc000071860, 0xc000012500)
	/home/pontus/govim/govim.go:238 +0x189
panic(0x650080, 0xc0007b01e0)
	/go/src/runtime/panic.go:522 +0x1b5
github.com/myitcv/govim/internal/plugin.Driver.Errorf(...)
	/home/pontus/govim/internal/plugin/plugin.go:61
github.com/myitcv/govim/internal/plugin.Driver.ChannelCall(0x6e3d40, 0xc000012500, 0x690996, 0x5, 0x691b4c, 0x9, 0xc000594140, 0x2, 0x2, 0xc000286280, ...)
	/home/pontus/govim/internal/plugin/plugin.go:75 +0x222
github.com/myitcv/govim/cmd/govim.(*vimstate).updateQuickfix(0xc0000a8640, 0x0, 0x0, 0x0, 0x10, 0xc000111f18)
	/home/pontus/govim/cmd/govim/quickfix.go:100 +0xf2c
github.com/myitcv/govim/cmd/govim.(*govimplugin).PublishDiagnostics.func1(0x6e3d40, 0xc000012500, 0xc000071860, 0xc000012500)
	/home/pontus/govim/cmd/govim/gopls_client.go:110 +0x62
github.com/myitcv/govim.(*govimImpl).Schedule.func1(0x0, 0x0)
	/home/pontus/govim/govim.go:247 +0x85
github.com/myitcv/govim.(*govimImpl).runEventQueue.func1(0x0, 0x6a1ab8)
	/home/pontus/govim/govim.go:567 +0x26
github.com/myitcv/govim.(*govimImpl).goHandleShutdown.func1(0x0, 0x0)
	/home/pontus/govim/govim.go:259 +0x54
gopkg.in/tomb%2ev2.(*Tomb).run(0xc000012590, 0xc00051e010)
	/home/pontus/gopath/pkg/mod/gopkg.in/tomb.v2@v2.0.0-20161208151619-d5d1b5820637/tomb.go:163 +0x2b
created by gopkg.in/tomb%2ev2.(*Tomb).Go
	/home/pontus/gopath/pkg/mod/gopkg.in/tomb.v2@v2.0.0-20161208151619-d5d1b5820637/tomb.go:159 +0xb7
</pre>

--- a/channel_cmds.go
+++ b/channel_cmds.go
@@ -4,8 +4,6 @@ import (
 	"encoding/json"
 	"fmt"
 	"io"
-
-	"gopkg.in/tomb.v2"
 )
 
 func (g *govimImpl) handleChannelError(ch unscheduledCallback, err error, format string, args ...interface{}) error {
@@ -126,9 +124,6 @@ func (g *govimImpl) DoProto(f func() error) (err error) {
 				}
 				err = r
 			case error:
-				if r == tomb.ErrDying {
-					panic(ErrShuttingDown)
-				}
 				if r == ErrShuttingDown {
 					panic(r)
 				}

--- a/event_queue.go
+++ b/event_queue.go
@@ -3,8 +3,6 @@ package govim
 import (
 	"encoding/json"
 	"fmt"
-
-	"gopkg.in/tomb.v2"
 )
 
 type eventQueueInst struct {
@@ -63,11 +61,11 @@ func (e eventQueueInst) handleUserQValueAndError(ch scheduledCallback, err error
 	args = append([]interface{}{}, args...)
 	select {
 	case <-e.govimImpl.tomb.Dying():
-		return nil, tomb.ErrDying
+		return nil, ErrShuttingDown
 	case e.flushEvents <- struct{}{}:
 		select {
 		case <-e.govimImpl.tomb.Dying():
-			panic(tomb.ErrDying)
+			return nil, ErrShuttingDown
 		case resp := <-ch:
 			if resp.errString != "" {
 				args = append(args, resp.errString)

--- a/govim.go
+++ b/govim.go
@@ -219,7 +219,7 @@ func (g *govimImpl) Schedule(f func(Govim) error) chan struct{} {
 	done := make(chan struct{})
 	g.eventQueue.Add(func() error {
 		defer func() {
-			if r := recover(); r != nil && r != ErrShuttingDown && r != tomb.ErrDying {
+			if r := recover(); r != nil && r != ErrShuttingDown {
 				panic(r)
 			}
 			close(done)
@@ -237,7 +237,7 @@ func (g *govimImpl) Schedule(f func(Govim) error) chan struct{} {
 func (g *govimImpl) goHandleShutdown(f func() error) {
 	g.tomb.Go(func() error {
 		defer func() {
-			if r := recover(); r != nil && r != ErrShuttingDown && r != tomb.ErrDying {
+			if r := recover(); r != nil && r != ErrShuttingDown {
 				panic(r)
 			}
 		}()
@@ -248,7 +248,7 @@ func (g *govimImpl) goHandleShutdown(f func() error) {
 func (g *govimImpl) load() error {
 	select {
 	case <-g.tomb.Dying():
-		return tomb.ErrDying
+		return ErrShuttingDown
 	case resp := <-g.unscheduledCallCallback("loaded"):
 		if resp.errString != "" {
 			return fmt.Errorf("failed to signal loaded to Vim: %v", resp.errString)
@@ -318,7 +318,7 @@ func (g *govimImpl) load() error {
 
 	select {
 	case <-g.tomb.Dying():
-		return tomb.ErrDying
+		return ErrShuttingDown
 	case resp := <-g.unscheduledCallCallback("initcomplete"):
 		if resp.errString != "" {
 			return fmt.Errorf("failed to signal initcomplete to Vim: %v", resp.errString)
@@ -434,7 +434,7 @@ func (g *govimImpl) run() error {
 					select {
 					case ch <- toSend:
 					case <-g.tomb.Dying():
-						return tomb.ErrDying
+						return ErrShuttingDown
 					}
 					return nil
 				})
@@ -443,7 +443,7 @@ func (g *govimImpl) run() error {
 					select {
 					case ch <- toSend:
 					case <-g.tomb.Dying():
-						return tomb.ErrDying
+						return ErrShuttingDown
 					}
 					return nil
 				})
@@ -543,7 +543,7 @@ GetWork:
 		if wait != nil {
 			select {
 			case <-g.tomb.Dying():
-				return tomb.ErrDying
+				return ErrShuttingDown
 			case <-wait:
 			}
 			continue GetWork
@@ -553,7 +553,7 @@ GetWork:
 		})
 		select {
 		case <-g.tomb.Dying():
-			return tomb.ErrDying
+			return ErrShuttingDown
 		case <-g.flushEvents:
 		}
 	}

--- a/internal/plugin/plugin.go
+++ b/internal/plugin/plugin.go
@@ -57,14 +57,17 @@ func (d Driver) do(f func() error) (err error) {
 	return f()
 }
 
-func (d Driver) Errorf(format string, args ...interface{}) {
+func (d Driver) errorf(check error, format string, args ...interface{}) {
+	if check == govim.ErrShuttingDown {
+		panic(check)
+	}
 	panic(ErrDriver{Underlying: fmt.Errorf(format, args...)})
 }
 
 func (d Driver) ChannelExpr(expr string) json.RawMessage {
 	i, err := d.Govim.ChannelExpr(expr)
 	if err != nil {
-		d.Errorf("ChannelExpr(%q) failed: %v", expr, err)
+		d.errorf(err, "ChannelExpr(%q) failed: %v", expr, err)
 	}
 	return i
 }
@@ -72,39 +75,39 @@ func (d Driver) ChannelExpr(expr string) json.RawMessage {
 func (d Driver) ChannelCall(name string, args ...interface{}) json.RawMessage {
 	i, err := d.Govim.ChannelCall(name, args...)
 	if err != nil {
-		d.Errorf("ChannelCall(%q) failed: %v", name, err)
+		d.errorf(err, "ChannelCall(%q) failed: %v", name, err)
 	}
 	return i
 }
 
 func (d Driver) ChannelEx(expr string) {
 	if err := d.Govim.ChannelEx(expr); err != nil {
-		d.Errorf("ChannelEx(%q) failed: %v", expr, err)
+		d.errorf(err, "ChannelEx(%q) failed: %v", expr, err)
 	}
 }
 
 func (d Driver) ChannelNormal(expr string) {
 	if err := d.Govim.ChannelNormal(expr); err != nil {
-		d.Errorf("ChannelNormal(%q) failed: %v", expr, err)
+		d.errorf(err, "ChannelNormal(%q) failed: %v", expr, err)
 	}
 }
 
 func (d Driver) ChannelRedraw(force bool) {
 	if err := d.Govim.ChannelRedraw(force); err != nil {
-		d.Errorf("ChannelRedraw(%v) failed: %v", force, err)
+		d.errorf(err, "ChannelRedraw(%v) failed: %v", force, err)
 	}
 }
 
 func (d Driver) Parse(j json.RawMessage, i interface{}) {
 	if err := json.Unmarshal(j, i); err != nil {
-		d.Errorf("failed to parse from %q: %v", j, err)
+		d.errorf(err, "failed to parse from %q: %v", j, err)
 	}
 }
 
 func (d Driver) ParseString(j json.RawMessage) string {
 	var v string
 	if err := json.Unmarshal(j, &v); err != nil {
-		d.Errorf("failed to parse string from %q: %v", j, err)
+		d.errorf(err, "failed to parse string from %q: %v", j, err)
 	}
 	return v
 }
@@ -112,7 +115,7 @@ func (d Driver) ParseString(j json.RawMessage) string {
 func (d Driver) ParseJSONArgSlice(j json.RawMessage) []json.RawMessage {
 	var v []json.RawMessage
 	if err := json.Unmarshal(j, &v); err != nil {
-		d.Errorf("failed to parse []json.RawMessage from %q: %v", j, err)
+		d.errorf(err, "failed to parse []json.RawMessage from %q: %v", j, err)
 	}
 	return v
 }
@@ -120,7 +123,7 @@ func (d Driver) ParseJSONArgSlice(j json.RawMessage) []json.RawMessage {
 func (d Driver) ParseInt(j json.RawMessage) int {
 	var v int
 	if err := json.Unmarshal(j, &v); err != nil {
-		d.Errorf("failed to parse int from %q: %v", j, err)
+		d.errorf(err, "failed to parse int from %q: %v", j, err)
 	}
 	return v
 }
@@ -128,7 +131,7 @@ func (d Driver) ParseInt(j json.RawMessage) int {
 func (d Driver) ParseUint(j json.RawMessage) uint {
 	var v uint
 	if err := json.Unmarshal(j, &v); err != nil {
-		d.Errorf("failed to parse int from %q: %v", j, err)
+		d.errorf(err, "failed to parse int from %q: %v", j, err)
 	}
 	return v
 }
@@ -143,19 +146,19 @@ func (d Driver) ChannelExf(format string, args ...interface{}) {
 
 func (d Driver) DefineFunction(name string, args []string, f DriverFunction) {
 	if err := d.Govim.DefineFunction(d.prefix+name, args, d.doFunction(f)); err != nil {
-		d.Errorf("failed to DefineFunction %q: %v", name, err)
+		d.errorf(err, "failed to DefineFunction %q: %v", name, err)
 	}
 }
 
 func (d Driver) DefineRangeFunction(name string, args []string, f DriverRangeFunction) {
 	if err := d.Govim.DefineRangeFunction(d.prefix+name, args, d.doRangeFunction(f)); err != nil {
-		d.Errorf("failed to DefineRangeFunction %q: %v", name, err)
+		d.errorf(err, "failed to DefineRangeFunction %q: %v", name, err)
 	}
 }
 
 func (d Driver) DefineCommand(name string, f DriverCommandFunction, attrs ...govim.CommAttr) {
 	if err := d.Govim.DefineCommand(d.prefix+name, d.doCommandFunction(f), attrs...); err != nil {
-		d.Errorf("failed to DefineCommand %q: %v", name, err)
+		d.errorf(err, "failed to DefineCommand %q: %v", name, err)
 	}
 }
 
@@ -164,14 +167,14 @@ func (d Driver) DefineAutoCommand(group string, events govim.Events, patts govim
 		group = strings.ToLower(d.prefix)
 	}
 	if err := d.Govim.DefineAutoCommand(group, events, patts, nested, d.doAutoCommandFunction(f), exprs...); err != nil {
-		d.Errorf("failed to DefineAutoCommand: %v", err)
+		d.errorf(err, "failed to DefineAutoCommand: %v", err)
 	}
 }
 
 func (d Driver) Viewport() govim.Viewport {
 	vp, err := d.Govim.Viewport()
 	if err != nil {
-		d.Errorf("failed to get Viewport: %v", err)
+		d.errorf(err, "failed to get Viewport: %v", err)
 	}
 	return vp
 }

